test/lazyexpression.jl

```diff
@@ -280,4 +290,45 @@ end
     @test @allocated(expr()) == 0
 end
 
+@testset "Fully-qualified names" begin
+    expr = @expression Base.:+(2, 2)
+    @test expr() == 4
+    @test @allocated(expr()) == 0
+
+    expr = @expression Base.:+(Base.:*(1, 2), 4 + 5)
+    @test expr() == 11
+    @test @allocated(expr()) == 0
+end
+
+@testset "Interpolated arguments" begin
+    e1 = @expression 1 + 2 * (3 * 4 + (5 * 6))
```

Comment:
Actually, this kind of thing should already just be eagerly evaluated straight away because of https://github.com/tkoolen/SimpleQP.jl/blob/e24789e660a5c99b1f0ac497ac92ca29e4697965/src/lazyexpression.jl#L60-L61

and

https://github.com/tkoolen/SimpleQP.jl/blob/e24789e660a5c99b1f0ac497ac92ca29e4697965/src/lazyexpression.jl#L66

but now that I think about it, that's currently broken because of the wrapping! Fixing that would also make your example of `@constraint model qnext <= collect_joint_position_upper_bounds(mechanism)` work.